src/treq/api.py

```diff
@@ -116,10 +119,19 @@ def request(method, url, **kwargs):
 
 def _client(*args, **kwargs):
     agent = kwargs.get('agent')
+    proxy = kwargs.get('proxy')
+    reactor = default_reactor(kwargs.get('reactor'))
+    pool = default_pool(reactor,
+                        kwargs.get('pool'),
+                        kwargs.get('persistent'))
+    if not agent and proxy is not None:
+        (address, port) = proxy
+        endpoint = HostnameEndpoint(
```

Comment:
what about authentication? This doesn't support proxy authentication if I'm not mistaken? or am I missing something?